cpu/msp430_common/startup.c

```diff
@@ -39,6 +39,10 @@ __attribute__((constructor)) static void startup(void)
     _init();
 #endif
 
+#ifdef MODULE_DBGPIN
+    dbgpin_init();
+#endif
```

Comment:
Why did you not put this before `board_init`, I personally tested on `z1` which in `board_init` does some manual configuration of gpio's, was this with that in mind?  But it would not allow using it in `msp430_cpu_init`.

Reply:
see comment above :-)

Reply:
but does the board_init on the z1 override the config for all pins? Then we might need to move the init...

Reply:
What comment? If its https://github.com/RIOT-OS/RIOT/pull/14304#issuecomment-692698786 doesn't it go in the opposite direction? there you argue for including `dbgpin` before `board_init`

> but does the board_init on the z1 override the config for all pins? 

Yep it sets a default for all pins, `z1_ports_init` is the specific function, but honestly, lets not optimize for that `BOARD`. Lets just put it where it makes more sense globally.

Reply:
I just notice, in the case of the msb430 I simply overlooked the `board_init` call above. I was just looking for the STDLIB init call and put it right after...

But as the board_init plays with pin configs, the location of the `dbgpin_init` should work as is best, right?

Reply:
So I think we have 3 msp430 CPU's that each have some custom calls to `ports_init`

- `telosb_ports_init(void)`
- `z1_ports_init(void)`
- `msb_ports_init(void)`

They all call `msp430_cpu_init` that calls `init_ports` that configures them all as I/O, but then that is overrided by the functions avbve..

So ideally we would call `dbgpin_init` after this `init_ports` and fix those custom initialization to only set the needed ports instead of all of them. But IMO this is out of scope so lets leave it as is.

Reply:
Maybe just move it before the `newlib` init

Reply:
done.